system-int-test/src/cmr/system_int_test/utils/subscription_util.clj

```diff
@@ -0,0 +1,152 @@
+   [cmr.umm-spec.versioning :as versioning]))
+
+(def versioned-content-type
+  "A versioned default content type used in the tests."
+  (mime-types/with-version
+   mime-types/umm-json versioning/current-subscription-version))
+
+(def default-ingest-opts
+  "Default HTTP client options for use when ingesting subscriptions using the functions below."
+  {:accept-format :json
+   :content-type mime-types/umm-json})
+
+(defn token-opts
+  "A little testing utility function that adds a user token to the default
+  headers (HTTP client options)."
+  [token]
+  (merge default-ingest-opts {:token token}))
+
+(defn grant-all-subscription-fixture
+  "A test fixture that grants all users the ability to create and modify subscriptions."
+  [f]
+  (echo-util/grant-all-subscription (s/context))
+  (f))
+
+(defn make-subscription-concept
+  "Convenience function for creating a subscription concept"
+  ([]
+   (make-subscription-concept {}))
+  ([metadata-attrs]
+   (make-subscription-concept metadata-attrs {}))
+  ([metadata-attrs attrs]
+   (make-subscription-concept metadata-attrs attrs 0))
+  ([metadata-attrs attrs idx]
+   (-> (merge {:provider-id "PROV1"} metadata-attrs)
+       (data-umm-sub/subscription-concept idx)
+       (assoc :format versioned-content-type)
+       (merge attrs))))
+
+(defn ingest-subscription
+  "A convenience function for ingesting a subscription during tests."
+  ([]
+   (ingest-subscription (make-subscription-concept)))
+  ([subscription-concept]
+   (ingest-subscription subscription-concept default-ingest-opts))
+  ([subscription-concept opts]
+   (let [result (ingest-util/ingest-concept subscription-concept opts)
+         attrs (select-keys subscription-concept
+                            [:provider-id :native-id :metadata])]
+     (merge result attrs))))
+
+(defn ingest-subscription-with-attrs
+  "Helper function to ingest a subscription with the given subscription attributes."
+  ([metadata-attrs]
+   (ingest-subscription (make-subscription-concept metadata-attrs)))
+  ([metadata-attrs attrs]
+   (ingest-subscription (make-subscription-concept metadata-attrs attrs)))
+  ([metadata-attrs attrs idx]
+   (ingest-subscription (make-subscription-concept metadata-attrs attrs idx))))
+
+(defn search-refs
+  "Searches for subscription using the given parameters and requesting the XML references format."
+  ([]
+   (search-refs {}))
+  ([params]
+   (search-refs params {}))
+  ([params options]
+   (search/find-refs :subscription params options)))
+
+(defn search-json
+  "Searches for subscription using the given parameters and requesting the JSON format."
+  ([]
+   (search-json {}))
+  ([params]
+   (search-json params {}))
+  ([params options]
+   (search/process-response
+    (transmit-search/search-for-subscriptions
+      (s/context) params (merge options
+                                {:raw? true
+                                 :http-options {:accept :json}})))))
+
+(defn subscription-result->xml-result
```

Comment:
Is this function used anywhere?

Reply:
probably not. but will be used later for the search I think. while I'm at the file, I just wanted to modify as many as I can.